File: Solutions/0198/0198.ts
function rob(nums: number[]): number {
    var prev1:number=0;
    var prev2:number=0;
    for(const num of nums){
        const dp:number=Math.max(prev1,prev2+num);
        prev2=prev1;
        prev1=dp;
    }
    return prev1;
};

File: Solutions/0198/0198.go
func rob(nums []int) int {
    prev1 := 0  // dp[i - 1]
    prev2 := 0  // dp[i - 2]

    for _, num := range nums {
        dp := max(prev1, prev2+num)
        prev2, prev1 = prev1, dp
    }

    return prev1
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}
